astroquery/utils/commons.py

```diff
@@ -467,3 +468,89 @@ def parse_votable(content):
     """
     tables = votable.parse(BytesIO(content), verify='warn')
     return tables
+
+
+def static_vars(**kwargs):
+    def decorate(func):
+        for k in kwargs:
+            setattr(func, k, kwargs[k])
+        return func
+    return decorate
+
+
+@static_vars(caps={})
+def get_access_url(service, reg_url, capability=None):
```

Comment:
Let this one shimmer a bit to see whether this method or something else would fit most use cases, including the cloud access we work on for Fornax with @zoghbi-a

Reply:
@bsipocz you seem to know what this is about - is this a more general pyvo-ish tool?  Does it belong here or there?

Reply:
Yes, that's part of the discussion of where to cut what belongs here and what in pyvo. Basically, my main point here is that access_url may not be disambigous, e.g. we're working on ways to have both the VO standard access_url, but also addresses for s3 bucket access for the same dataset. 

Anyway, this will be more like some big picture API discussion, I wouldn't yet go down to the nitpicky detailed review for a draft PR yet, but work out the scope first.

Reply:
There is a good argument for putting this into `pyvo`.  It's specific to reading the properties document from an IVOA Registry service, as well as the XML Capabilities (https://www.ivoa.net/documents/DALI/20170517/REC-DALI-1.1.html#tth_sEc2.5) document returned from an IVOA web service.  Is the term `access url` overused already?

Reply:
Thank you @bsipocz .  It is a near timeline to adapt to the latest DataLink service which will be out soon.  I can put this into the `alma/core.py` for the time being, which is what `cadc` does.  The idea behind putting it into a common location would be to share it with `cadc` as well.

Reply:
yes, I would love to have a shared way to do it, but ideally, it should also work for mast, and irsa (I'm refactoring it to use pyvo), and heasarch/direct users of pyvo.